sympy/calculus/util.py

```diff
@@ -1348,20 +1349,66 @@ def __pow__(self, other):
                         return AccumBounds(self.min**other, self.max**other)
 
                 num, den = other.as_numer_denom()
-                if num == S.One:
+                if num is S.One:
                     if den % 2 == 0:
-                        if S.Zero in self:
-                            if self.min.is_extended_negative:
-                                return AccumBounds(0, real_root(self.max, den))
-                    return AccumBounds(real_root(self.min, den),
-                                       real_root(self.max, den))
-                if den!=1:
-                    num_pow = self**num
-                    return num_pow**(1 / den)
+                        try:
+                            if S.Zero in self:
+                                return AccumBounds(
+                                    real_root(self.min, den),
+                                    real_root(self.max, den))
+                        except TypeError:
+                            AccumBounds(-oo, oo)
+                elif den is not S.One:  # e.g. if other is not Float
+                    return (self**num)**(1/den)  # ok for non-negative base
+
+            if isinstance(other, AccumBounds):
+                if (
+                        self.min.is_zero is not None and
+                        other.min.is_negative is not None):
+                    # negative numbers require only integer exponents to produce
+                    # real bounds and AccumBounds includes non-integers so we
+                    # must deal only with nonnegative bases in self.
+                    p = set()
+                    if self.min.is_zero:
+                        if other.min.is_extended_negative:
+                            raise ZeroDivisionError
+                        if other.max.is_extended_positive:
+                            p.add(0)
+                    if 0 in other:
+                        p.add(1)  # x**0 = 1
+                    if 1 in self:
+                        p.add(1)  # 1**x = 1
+                    b = set(self.args) - {0}  # handled above
+                    e = set(other.args)
+                    # compute candidate powers
+                    p |= {i**j for i in b for j in e}
+                    return self.func(min(p), max(p))
+
             return AccumBounds(-oo, oo)
 
         return NotImplemented
 
+    @_sympifyit('other', NotImplemented)
+    def __rpow__(self, other):
+        if other.is_real and other.is_nonnegative:
+            if other is S.One:
+                return S.One
+            p = set()
+            if other.is_extended_positive:
+                p.add(other**self.min)
+                p.add(other**self.max)
+            elif other.is_zero:
+                if self.min.is_zero:
+                    p.add(0)
+                    if self.max.is_extended_positive:
+                        p.add(1)
+                elif self.min.is_extended_positive:
+                    p.add(1)
```

Comment:
Should I raise `ZeroDivisionError` if `self.min.is_extended_nonpositive`?

Reply:
`0**negative` returns `zoo`. Maybe we could avoid raising error and return `zoo` if also `self.max.is_extended_nonpositive`. Otherwise I think that `nan` could be returned

Reply:
I will try just an unevaluated expression since the user could then substitute in a value later.